package/MDAnalysis/analysis/diffusionmap.py

```diff
@@ -0,0 +1,215 @@
+   >>> eg,ev=diffusionmap.diffusionmap(u)
+
+To see how the two slowest collective coordinates how the Other stuff in paper
+
+   >>> import matplotlib.pyplot as plt
+   >>> import matplotlib.pyplot as plt
+   >>> plt.scatter(ev[:,1],ev[:,2])
+   >>> plt.title('diffusion map')
+   >>> plt.xlabel("slowest collective coordinate")
+   >>> plt.ylabel("second slowest collective coordinate")
+   >>> plt.show()
+
+
+Common usage
+------------
+
+To **fit a single structure** with :func:`alignto`::
+
+   >>> ref = Universe(PSF, PDB_small)
+   >>> mobile = Universe(PSF, DCD)     # we use the first frame
+   >>> alignto(mobile, ref, select="protein and name CA", mass_weighted=True)
+
+This will change *all* coordinates in *mobile* so that the protein
+C-alpha atoms are optimally superimposed (translation and rotation).
+
+
+Functions
+---------
+
+.. autofunction:: diffusionmap
+
+
+"""
+
+import os.path
+from six.moves import range, zip, zip_longest
+import numpy as np
+import warnings
+import logging
+
+
+import MDAnalysis.lib.qcprot as qcp
+from MDAnalysis.exceptions import SelectionError, SelectionWarning
+from MDAnalysis.lib.log import ProgressMeter
+import MDAnalysis.analysis.rms as rms
+
+
+logger = logging.getLogger('MDAnalysis.analysis.diffusionmap')
+
+
+def diffusionmap(u, select='all', epsilon='average', k=10, weight=None):
+    """ Non-linear dimension reduction method diffusion map
+    
+    The dimension reduction is done in the following way:
+
+    1. A RMSD between each every pair of frames is calculated.
+    2. The normalized kernel is obtain as in [1] and [2].
+    3. The eigenvalues and eigenvectors of the normalized kernel are the output.
+
+    :Arguments:
+      *u*
+         trajectory :class:`~MDAnalysis.core.AtomGroup.Universe`
+         The trajectory can be a long trajectory 
+      *select*
+         1. any valid selection string for
+            :meth:`~MDAnalysis.core.AtomGroup.AtomGroup.select_atoms` 
+         This selection of atoms is used to calculate the RMSD between different frames. Water should be excluded.
+      *epsilon* : float 
+          Specifies the epsilon used for the diffusion map. More information in [1] and [2] 
+          With 'average' the average of the RMSD to the k-nearest-neighbor will be used.  
+      *k* : specifies the k for the k-nearest-neighbor is average epsilon is used.
+      *weight* : None or numpy array of the same length as the trajectory
+         With 'None' the weight of each frame of the trajectory will be the same.
+         If order of the weights has to be the same as the order of framesin the trajectory.
+
+
+    :Returns: eigenvalues, eigenvectors 
+    the second and higher eigenvectors ev[i+1,:] represent the i-th slowest collective coordinates.
+
+    """
+
+    frames=u.trajectory
+    ref_atoms = u.select_atoms(select)
+    nframes=len(frames)
+    natoms = ref_atoms.n_atoms
+
+    rmsd_matrix=np.zeros((nframes,nframes), dtype=np.float64)
```

Comment:
Style: follow PEP8, in particular spaces around `=` assignments, space after commas, etc. See links under [Style Guide: Coding Style](https://github.com/MDAnalysis/mdanalysis/wiki/Style-Guide#coding-style).